dora/core/server/worker/src/main/java/alluxio/worker/s3/S3HttpHandler.java

```diff
@@ -68,47 +73,71 @@ public void channelReadComplete(ChannelHandlerContext ctx) throws Exception {
-          S3NettyHandler.createHandler(context, request, mFileSystem, mDoraWorker,
-              mAsyncAuditLogWriter);
-    } catch (Exception ex) {
-      LOG.error("Exception during create s3handler: ", ex);
-      HttpResponse errorResponse = S3ErrorResponse.createNettyErrorResponse(ex, request.uri());
-      ChannelFuture f = context.writeAndFlush(errorResponse);
-      f.addListener(ChannelFutureListener.CLOSE);
-      return;
-    }
-
-    try {
-      // Handle request async
-      if (Configuration.getBoolean(PropertyKey.WORKER_S3_ASYNC_PROCESS_ENABLED)) {
-        FullHttpRequest asyncRequest = request.copy();
-        S3NettyBaseTask.OpTag opTag = s3Handler.getS3Task().mOPType.getOpTag();
-        ExecutorService es = (ExecutorService) (opTag == S3NettyBaseTask.OpTag.LIGHT
-            ? mLightPool : mHeavyPool);
+      HttpResponse response = null;
+      if (msg instanceof HttpRequest) {
+        HttpRequest request = (HttpRequest) msg;
+        try {
+          mHandler =
+              S3NettyHandler.createHandler(context, request, mFileSystem, mDoraWorker,
+                  mAsyncAuditLogWriter);
+        } catch (Exception ex) {
+          LOG.error("Exception during create s3handler: ", ex);
+          HttpResponse errorResponse = S3ErrorResponse.createNettyErrorResponse(ex, request.uri());
+          ChannelFuture f = context.writeAndFlush(errorResponse);
+          f.addListener(ChannelFutureListener.CLOSE);
+          return;
+        }
+        // Handle request async
+        if (!mHandler.getS3Task().getOPType().equals(S3NettyBaseTask.OpType.PutObject)
+            && Configuration.getBoolean(PropertyKey.WORKER_S3_ASYNC_PROCESS_ENABLED)) {
+          S3NettyBaseTask.OpTag opTag = mHandler.getS3Task().mOPType.getOpTag();
+          ExecutorService es =
+              (ExecutorService) (opTag == S3NettyBaseTask.OpTag.LIGHT ? mLightPool : mHeavyPool);
 
-        es.submit(() -> {
-          try {
-            HttpResponse response = s3Handler.getS3Task().continueTask();
-            s3Handler.processHttpResponse(response);
-          } catch (Throwable th) {
-            HttpResponse errorResponse =
-                S3ErrorResponse.createNettyErrorResponse(th, asyncRequest.uri());
-            ChannelFuture f = context.writeAndFlush(errorResponse);
-            f.addListener(ChannelFutureListener.CLOSE);
-          }
-        });
+          es.submit(() -> {
+            try {
+              HttpResponse asyncResponse = mHandler.getS3Task().continueTask();
+              mHandler.processHttpResponse(asyncResponse);
+            } catch (Throwable th) {
+              HttpResponse errorResponse =
+                  S3ErrorResponse.createNettyErrorResponse(th, request.uri());
+              ChannelFuture f = context.writeAndFlush(errorResponse);
+              f.addListener(ChannelFutureListener.CLOSE);
+            }
+          });
+          return;
+        }
+        response = mHandler.getS3Task().continueTask();
```

Comment:
is this part of the handle req async block?

Reply:
this line is for handle HttpRequest Synchronously